util/file.py

```diff
@@ -626,3 +630,26 @@ def find_broken_symlinks(rootdir, followlinks=False):
                     broken_links_to_remove.append(fpath.rstrip("/"))
 
     return broken_links_to_remove
+
+
+def join_interleaved_fastq(input_fastq_f, output_format='fastq', num_n=None):
+    '''Join interleaved fastq into single reads connected by Ns'''
+    assert output_format in ('fastq', 'fasta')
+    num_n = num_n or 31
+
+    # In case input is a stdout pipe (always binary encoded)
+    input_fastq_f = codecs.getreader('ascii')(input_fastq_f)
+    for r1, r2 in util.misc.batch_iterator(SeqIO.parse(input_fastq_f, 'fastq'), 2):
+        if r1.id.endswith('/1'):
+            rid = r1.id[:-2]
+        else:
+            rid = r1.id
+        jseq = Seq(str(r1.seq) + 'A' * num_n + str(r2.seq))
+        labbrevs = None
+        if output_format == 'fastq':
+            # Illumina quality score of 2 indicates unreliable base
+            labbrevs = {
+                'phred_quality': r1.letter_annotations['phred_quality'] + [2] * num_n + r2.letter_annotations['phred_quality']
+            }
+        rec = SeqRecord(jseq, id=rid, description='', letter_annotations=labbrevs)
```

Comment:
If you're really into speed optimization, I've heard it said that creating and destroying a ton of BioPython SeqRecord objects for ephemeral use can be somewhat inefficient (if you're not translating or revcomping or doing other biological things to it) ... might consider just emitting a tuple of strings instead?